templates/addons/env/s3.yml

```diff
@@ -35,15 +35,16 @@ Resources:
         - Key: Copilot-application
           Value: !Ref App
         - Key: Copilot-environment
-          Value: !Ref Env 
+          Value: !Ref Env
         - Key: Copilot-name
-          Value: !Ref Name 
+          Value: !Ref Name
 
   {{ service.prefix }}BucketPolicy:
     Metadata:
       'aws:copilot:description': 'A bucket policy to deny unencrypted access to the bucket and its contents'
     Type: AWS::S3::BucketPolicy
     DeletionPolicy: Retain
+    UpdateReplacePolicy: Retain
```

Comment:
See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatereplacepolicy.html

Reply:
Would be good to get your eyes on this one in particular before I merge @lgarvey

Reply:
So s3 buckets names are unique. In this instance the bucket would be retained, and then if we attempt to recreate the env it would fail due to the existing bucket name.  That doesn't seem to be an issue though - in this instance, they could replace the s3 type with an s3-policy so that access to the bucket is maintained.